File: mvvmbase/src/main/java/com/bing/mvvmbase/base/viewpager/BaseFragmentPagerAdapter.java
package com.bing.mvvmbase.base.viewpager;

import java.util.List;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

public class BaseFragmentPagerAdapter  extends FragmentPagerAdapter {
	protected List<Fragment> mFragments;

	public BaseFragmentPagerAdapter(FragmentManager fm, List<Fragment> fragments) {
		super(fm);
		mFragments = fragments;
	}

	@Override
	public Fragment getItem(int position) {
		return mFragments.get(position);
	}

	@Override
	public int getCount() {
		return mFragments.size();
	}
}

File: mvvmbase/src/main/java/com/bing/mvvmbase/base/viewpager/BaseFragmentPagerAdapter.kt
package com.bing.mvvmbase.base.viewpager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class BaseFragmentPagerAdapter(fm: FragmentManager, protected var mFragments: List<Fragment>) : FragmentPagerAdapter(fm) {

        override fun getItem(position: Int): Fragment {
                return mFragments[position]
        }

        override fun getCount(): Int {
                return mFragments.size
        }
}
